apps/api/src/middleware/__tests__/auth.middleware.test.ts

```diff
@@ -0,0 +1,587 @@
+import { Request, Response } from "express";
+import { MockedObject, beforeEach, describe, expect, it, vi } from "vitest";
+
+import {
+  SelectAdmin,
+  SelectAgent,
+  SelectUser,
+} from "@recallnet/db/schema/core/types";
+
+import { authMiddleware } from "@/middleware/auth.middleware.js";
+import { ApiError } from "@/middleware/errorHandler.js";
+import { AdminService } from "@/services/admin.service.js";
+import { AgentService } from "@/services/agent.service.js";
+import { UserService } from "@/services/user.service.js";
+
+// Mock dependencies
+vi.mock("@/lib/logger.js", () => {
+  const mockLogger = {
+    debug: vi.fn(),
+    info: vi.fn(),
+    warn: vi.fn(),
+    error: vi.fn(),
+  };
+  return {
+    authLogger: mockLogger,
+  };
+});
+
+vi.mock("@/lib/privy/utils.js", () => ({
+  extractPrivyIdentityToken: vi.fn(),
+}));
+
+vi.mock("@/lib/privy/verify.js", () => ({
+  verifyPrivyIdentityToken: vi.fn(),
+}));
+
+vi.mock("@/middleware/auth-helpers.js", () => ({
+  extractApiKey: vi.fn(),
+  isLoginEndpoint: vi.fn(),
+}));
+
+describe("authMiddleware", () => {
+  let mockAgentService: MockedObject<AgentService>;
+  let mockUserService: MockedObject<UserService>;
+  let mockAdminService: MockedObject<AdminService>;
+  let middleware: ReturnType<typeof authMiddleware>;
+  let mockReq: Partial<Request>;
+  let mockRes: Partial<Response>;
+  let mockNext: ReturnType<typeof vi.fn>;
+
+  beforeEach(async () => {
+    vi.clearAllMocks();
+
+    // Create mock services
+    mockAgentService = {
+      validateApiKey: vi.fn(),
+      getAgent: vi.fn(),
+    } as unknown as MockedObject<AgentService>;
```

Comment:
To avoid this mess (that the AIs seem to love to create), we're finding it super useful to use this package https://github.com/eratio08/vitest-mock-extended

@asutula used it here https://github.com/recallnet/js-recall/blob/main/apps/api/src/services/__tests__/boost-award.service.test.ts to really good effect, and it makes mocking type safe, none of the `as any` or `as unknown as X` nonsense.